### PR TITLE
Added densitycurrent.jl at 100m resolution

### PR DESCRIPTION
**Pull request by smarras79**
# Description

* This is an updated replacement of [#1151](https://github.com/CliMA/ClimateMachine.jl/pull/1151/). Uses Vreman model. 
* This test runs to completion in 3 minutes of wallclock.
* Updates turbulence model to correct multiplier bug-fix. 

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by charleskawczynski** [comment]
FYI, the rising bubble tutorial was updated in #1236

**Comment by akshaysridhar** [comment]
> FYI, the rising bubble tutorial was updated in #1236

@charleskawczynski removed the changes to rtb from this given #1236

**Comment by smarras79** [comment]
@charleskawczynski I see that the following have been removed from the list of tutorials.
These are all supposed to be as part of the tutorials. The markdown have been fixed in the agnesi* files, density current, and risingbubble. 

```
"Atmos" => [                                                                                                                           
            "Dry Rayleigh Bernard" => "Atmos/dry_rayleigh_benard.jl",                                                                          
            "Dry Idealized GCM" => "Atmos/heldsuarez.jl",                                                                                      
            "Rising Thermal Bubble" => "Atmos/risingbubble.jl",                                                                                
            "Linear Hydrostatic Mountain" => "Atmos/agnesi_hs_lin.jl",                                                                         
            "Linear Non-Hydrostatic Mountain" => "Atmos/agnesi_nh_lin.jl",   
```

**Comment by akshaysridhar** [comment]
bors try

File: src/Atmos/Model/turbulence.jl [hunk 1]
```diff
@@ -517,7 +517,7 @@ function turbulence_tensors(
     Richardson = diffusive.turbulence.N² / (normS^2 + eps(normS))
     f_b² = sqrt(clamp(1 - Richardson * _inv_Pr_turb, 0, 1))
 
-    β = f_b² * (aux.turbulence.Δ)^2 * (α' * α)
+    β = (aux.turbulence.Δ)^2 * (α' * α)
```
Comment (kpamnany) [inline src/Atmos/Model/turbulence.jl:520]: This change should be documented in the PR or have a commit just for it.

Comment (akshaysridhar) [reply src/Atmos/Model/turbulence.jl:520]: @smarras79 can split the commit if he chooses but the notes have been updated.

**Comment by charleskawczynski** [comment]
Yep, we can add the tutorials back in once they generate valid markdown and don’t take too long.

**Comment by charleskawczynski** [comment]
Can we fix the broken links in this tutorial? All the links in the preview should work, and many give 404 errors.

**Comment by smarras79** [comment]
bors+

**Comment by smarras79** [comment]
bors-

**Comment by smarras79** [comment]
> Can we fix the broken links in this tutorial? All the links in the preview should work, and many give 404 errors.

which ones and what should they be replaced with?

**Comment by charleskawczynski** [comment]
> which ones and what should they be replaced with?

Click on each link [here](https://clima.github.io/ClimateMachine.jl/previews/PR1237/generated/Atmos/densitycurrent/#EX-DC-docs-1). The broken ones I see are
 - AtmosModel (occurs several times)
 - FlatOrientation
 - CLIMAParameters.jl
 - MoistThermodynamics

But there are several more.

**Comment by smarras79** [comment]
bors+

**Comment by smarras79** [comment]
@charleskawczynski could you please check why this is not passing the documentation test and merge it? It needs to be merged as soon as possible. Thanks

**Comment by charleskawczynski** [comment]
> @charleskawczynski could you please check why this is not passing the documentation test and merge it? It needs to be merged as soon as possible. Thanks

Our docs have a 60 min limit, and it's timing out. Maybe #1250 will help. I'll try to run this tutorial locally to see how long it runs.

**Comment by smarras79** [comment]
> > @charleskawczynski could you please check why this is not passing the documentation test and merge it? It needs to be merged as soon as possible. Thanks
> 
> Our docs have a 60 min limit, and it's timing out. Maybe #1250 will help. I'll try to run this tutorial locally to see how long it runs.

Thanks for helping, @charleskawczynski  Could you also fix #1163 at once so that we can finally have those tutorials in the code and not think about it. Thanks.
It would be useful to have a documentation template for tutorials so that we all use that and simply modify the text and information.

File: tutorials/Atmos/densitycurrent.jl [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    ## state for the linearisation step is also defined.
+    T_surface = FT(300)
+    T_min_ref = FT(0)
+    T_profile = DryAdiabaticProfile{FT}(param_set, T_surface, T_min_ref)
+    ref_state = HydrostaticState(T_profile)
+
+    ## The fun part! Here we assemble the `AtmosModel`.
+    ##md # !!! note
+    ##md #     Docs on model subcomponent options can be found here:
+    ##md #     - [`param_set`](https://CliMA.github.io/CLIMAParameters.jl/latest/)
```
Comment (charleskawczynski) [inline tutorials/Atmos/densitycurrent.jl:201]:
```suggestion
    ##md #     - [`param_set`](https://CliMA.github.io/CLIMAParameters.jl/dev/)
```

File: tutorials/Atmos/densitycurrent.jl [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+    ## Choose an Explicit Single-rate Solver LSRK144 from the existing [ODESolvers](@ref
+    ## ODESolvers-docs) options Apply the outer constructor to define the
+    ode_solver = ClimateMachine.ExplicitSolverType(
+        solver_method = LSRK144NiegemannDiehlBusch,
+    )
+
+    ## The model coefficient for the turbulence closure is defined via the
+    ## [CLIMAParameters
+    ## package](https://CliMA.github.io/CLIMAParameters.jl/latest/) A reference
```
Comment (charleskawczynski) [inline tutorials/Atmos/densitycurrent.jl:191]:
```suggestion
    ## package](https://CliMA.github.io/CLIMAParameters.jl/dev/) A reference
```

File: tutorials/Atmos/densitycurrent.jl [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        turbulence = Vreman(_C_smag),                   ## Turbulence closure model
+        moisture = DryModel(),                          ## Exclude moisture variables
+        source = (Gravity(),),                          ## Gravity is the only source term here
+        tracers = NoTracers(),                          ## Tracer model with diffusivity coefficients
+        ref_state = ref_state,                          ## Reference state
+        init_state_conservative = init_densitycurrent!, ## Apply the initial condition
+    )
+
+    ## Finally, we pass a `Problem Name` string, the mesh information, and the
+    ## model type to  the [`AtmosLESConfiguration`] object.
```
Comment (charleskawczynski) [inline tutorials/Atmos/densitycurrent.jl:220]:
```suggestion
    ## model type to  the [`AtmosLESConfiguration`](@ref ClimateMachine.AtmosLESConfiguration) object.
```

**Comment by charleskawczynski** [comment]
On my machine, this tutorial is taking ~50 min. The docs are not run on GPUs, so we need to trim down the runtime for this tutorial (to maybe 5-10 min). I suggest coarsening the spatial/temporal resolution. Also, we can always add comments to suggest higher resolution for users.

**Comment by smarras79** [comment]
we can set timeend = FT(100) instead of FT(1000). That should do

**Comment by charleskawczynski** [comment]
I'll add a few commits

**Comment by akshaysridhar** [comment]
bors try

**Comment by charleskawczynski** [comment]
@akshaysridhar can you add this tutorial to the pages array?

**Pull request merged**